src/main/java/vn/triplet/controller/web/ProductController.java

```diff
@@ -53,12 +61,15 @@ public String showProductDetails(
 		}
 		List<Product> similarProducts = productService.loadProductWithCategoryId(product.getCategory().getId(), product.getId());
 		LinkedList<Integer> recentlyViewedProductIds = (LinkedList<Integer>)session.getAttribute("recentlyViewedProductIds");
-		List<Product> recentlyViewedProducts = productService.loadProductWithListProductId((List<Integer>)recentlyViewedProductIds);
+		List<Product> recentlyViewedProducts = productService.loadProductWithListProductId((List<Integer>)recentlyViewedProductIds);		
+		List<Rate> ratings=rateService.loadRatings(id);
+		RateInfo rate= new RateInfo();
 		
 		model.addAttribute("product", product);
 		model.addAttribute("similarProducts", similarProducts);
 		model.addAttribute("recentlyViewedProducts", recentlyViewedProducts);
-		
+		if(ratings.size()>0)model.addAttribute("ratings",ratings);
```

Comment:
Chị thấy đâu cần trả lại một mớ rating thế này, Product chỉ quan tâm rating trung binhg của nó là bao nhiu thôi nè :p 
Nên chị nghĩ chỉ cần số ng rating (ratings.count) và số rating trung bình là bao nhiu sao, add vào trong objedt product infor là ok nef



khi nào cần bấm vào ratings detail of product thì mới cần view chi tiết là bao nhìu ng rating, rate mấy sao

Reply:
ok ạ

Reply:
dạ, tại tụi em làm trong rating có cả comment luôn á chị, nên muốn load ra hết cả comment với value rating luôn ạ

Reply:
ok ok